### PR TITLE
Add wildcard CDI test.

### PR DESCRIPTION
**Pull request by alesj**
@mkouba I guess this config should be valid?

### CODE REVIEW

File: independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/producer/generic/WildcardProducerHierarchyTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+        @Produces
+        @ApplicationScoped
+        public AsyncBiFunctionService.WithInner<String, Long, Date> inner() {
+            return new InnerImpl();
+        }
+
+        @Produces
+        @Singleton
+        public LocalService<? extends AsyncBiFunctionService.WithInner<String, Long, Date>> localInner(AsyncBiFunctionService.WithInner<String, Long, Date> service) {
```
Comment (mkouba) [inline independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/producer/generic/WildcardProducerHierarchyTest.java]: `LocalService<? extends AsyncBiFunctionService.WithInner<String, Long, Date>>` is not a legal bean type and should be removed from the set of bean types of this producer method.

Comment (alesj) [reply independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/producer/generic/WildcardProducerHierarchyTest.java]: Why is that?

Comment (mkouba) [reply independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/producer/generic/WildcardProducerHierarchyTest.java]: https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#legal_bean_types

**Comment by manovotn** [comment]
@alesj @mkouba FTR here is the same test using type variables - https://github.com/manovotn/quarkus/tree/alesj-wild_cdi_1_versionWithTypeVar
And that one works as expected.

**Comment by manovotn** [comment]
Closing this PR as the behavior is spec defined.

**Pull request closed without merging**